client/scripts/app.ts

```diff
@@ -61,8 +61,11 @@ export async function initAppState(updateSelectedNode = true): Promise<void> {
       app.config.notificationCategories = data.notificationCategories
         .map((json) => NotificationCategory.fromJSON(json));
       app.config.invites = data.invites;
-      app.recentActivity.activeAddresses = data.activeAddresses;
-      app.recentActivity.activeThreads = data.activeThreads;
+
+      // add recentActivity
+      const { recentThreads, recentComments } = data;
+      app.recentActivity.addThreads(recentThreads);
+      app.recentActivity.addAddressesFromActivity(recentThreads.concat(recentComments));
```

Comment:
Clear before adding. Otherwise we will end up with duplicates in the store (try going through web3 login).